### DISCUSSION TITLE
Programatically set image width and height when uploading to external source

### DISCUSSION THREAD

**Discussion by lhguerra** [JiHong88/SunEditor#1109] [Q&A] [answered]
I'm using sun editor and I've added a callback to upload the images to Cloudinary when using the image button in the editor. It works just fine, same as the example, but I need something else: the markup generated when inserting the image into the content should have width and height properties. I have the width and height from the return of the cloudinary request, so it is just a matter of programatically making it be printed in the HTML output.

Here is my onImageUploadBefore script:
```
  editor.onImageUploadBefore = (files, info, core, uploadHandler) => {
    const attachment = files[0];
    const formData = new FormData();
    formData.append('image', attachment);

    $.ajax({
      type: 'POST',
      data: formData,
      contentType: false,
      processData: false,
      url: `/rich_text_image_upload`,
      beforeSend: function(xhr) {xhr.setRequestHeader('X-CSRF-Token', $('meta[name="csrf-token"]').attr('content'))},
      success: (data) => {
        const uploaded_image = data.uploaded_image
        const response = {
          "result": [
            {
              "url": uploaded_image.secure_url,
              "name": `${uploaded_image.secure_url.original_filename}.${uploaded_image.format}`,
              "size": uploaded_image.bytes
            },
          ]
        };
        uploadHandler(response);
      },
      fail: () => alert("Image upload failed")
    })
  }
});
```
First I thought that some prop in the result object should do it, but it looks like I'm not changing the correct place.

**Comment by JiHong88** [comment, marked as answer]
Before calling "uploadHandler", the property value of the "info" object must be changed.
```
editor.onImageUploadBefore = (files, info, core, uploadHandler)=>{
  info.inputWidth=''
  info.inputHeight=''
}
```
http://suneditor.com/sample/html/out/document-user.html#onImageUploadBefore

**Reply by lhguerra** [reply]
Thank's, it worked like a charm!
